packages/js-sdk/src/sandbox/filesystem/index.ts

```diff
@@ -183,11 +171,11 @@ export class Filesystem {
     }
 
     const files = res.data
-    if (!files || files.length === 0) {
+    if (!files) {
       throw new Error('Expected to receive information about written file')
     }
 
-    return files[0] as EntryInfo
+    return files.length === 1 && path ? (files[0] as EntryInfo) : (files as EntryInfo[])
```

Comment:
Right not the type of files/files[0] is structurally the same as the `EntryInfo` and we are relying on that without explicitly checking. This is also not blocking right now, but something I wanted to point out.

Reply:
In the meanwhile i moved the `as` up to `res.data`